pkg/machines/components/vm/memoryModal.jsx

```diff
@@ -0,0 +1,235 @@
+                                    });
+                        }
+                    });
+        } else if (vm.currentMemory !== this.state.memory) {
+            dispatch(setMemory(vm, this.state.memory))
+                    .fail(exc => this.dialogErrorSet(_("Memory could not be saved"), exc.message))
+                    .then(() => {
+                        if (vm.state !== 'running')
+                            dispatch(getVm({ connectionName: vm.connectionName, id: vm.id }));
+                        this.close();
+                    });
+        } else {
+            this.close();
+        }
+    }
+
+    render() {
+        const vm = this.props.vm;
+        const idPrefix = vmId(vm.name) + '-memory-modal';
+        const defaultBody = (
+            <div id='memory-config-dialog' className='ct-form-layout'>
+                <label className='control-label'>
+                    {_("Current Allocation")}
+                </label>
+                <div className='form-group ct-validation-wrapper'>
+                    <div role='group'>
+                        <input id={`${idPrefix}-memory`}
+                            className='form-control'
+                            type='number'
+                            value={toFixedPrecision(convertToUnit(this.state.memory, 'KiB', this.state.memoryUnit))}
+                            step={1}
+                            onChange={e => this.onValueChanged('memory', e.target.value)}
+                            onClick={e => { // releasing arrows does not trigger a seperate on blur event
+                                this.onValueChanged('memory', e.target.value);
+                                this.onValueBlurred('memory', e.target.value);
+                            }}
+                            onBlur={e => this.onValueBlurred('memory', e.target.value)} />
+                        <Select.Select id={`${idPrefix}-memory-unit`}
+                            initial={this.state.memoryUnit}
+                            onChange={value => this.onUnitsValueChanged('memory', value)}>
+                            <Select.SelectEntry data={units.MiB.name} key={units.MiB.name}>
+                                {_("MiB")}
+                            </Select.SelectEntry>
+                            <Select.SelectEntry data={units.GiB.name} key={units.GiB.name}>
+                                {_("GiB")}
+                            </Select.SelectEntry>
+                        </Select.Select>
+                    </div>
+                    <HelpBlock>
+                        {_("Memory can be adjusted between 128 MB and the maximum allocation.")}
+                    </HelpBlock>
+                </div>
+
+                <hr />
+
+                <label className='control-label'>
+                    {_("Maximum Allocation")}
+                </label>
+                <div className='form-group ct-validation-wrapper'>
+                    <div role='group'>
```

Comment:
These two `<div>`s should probably be combined into one, if possible.

`<div role="group" className="form-group ct-validation-wrapper">`

We might need this changed elsewhere in the file too.

(If it's not possible, I might need to adjust ct-form-layout even more. Hmm.)

Reply:
It's not possible, because the div with role group should not contain the div with help block.
Otherwise they will end up in the same line.